### PR TITLE
Improve documentation

### PR DESCRIPTION
**Pull request by tekktrik**
A cleaned up version of #22 with all the changes of #25 separated

### CODE REVIEW

**Comment by tekktrik** [comment]
Requires `pyasn1` in `requirements.txt` so that Sphinx will work properly, but I don't think it will install when the library is installed via pip because it isn't in `setup.py`

File: adafruit_rsa/key.py [hunk 1]
```diff
@@ -45,58 +54,51 @@ class AbstractKey(object):
...
         """Saves the key in PKCS#1 DER format, implement in a subclass.
 
-        :returns: the DER-encoded key.
+        :return: the DER-encoded key.
         :rtype: bytes
         """
 
     @classmethod
-    def load_pkcs1(cls, keyfile, format="PEM"):
+    def load_pkcs1(cls, keyfile: bytes, format: str = "PEM") -> "AbstractKey":
```
Comment (FoamyGuy) [inline adafruit_rsa/key.py]: is `format` a good candidate for doing the `Literal["PEM", "DER"]` type? It looks like those are the two valid values from what I can tell.

File: adafruit_rsa/key.py [hunk 2]
```diff
@@ -124,12 +128,11 @@ def _assert_format_exists(file_format, methods):
                 "Unsupported format: %r, try one of %s" % (file_format, formats)
             ) from err
 
-    def save_pkcs1(self, format="PEM"):
+    def save_pkcs1(self, format: str = "PEM") -> bytes:
```
Comment (FoamyGuy) [inline adafruit_rsa/key.py]: possibly another spot for the Literal type thing. Although thinking more about it, I'm not sure if that works for strings. Do you happen to know that @tekktrik?

File: adafruit_rsa/key.py [hunk 3]
```diff
@@ -735,28 +741,31 @@ def gen_keys(nbits, getprime_func, accurate=True, exponent=DEFAULT_EXPONENT):
 
 
 def newkeys(
-    nbits, accurate=True, poolsize=1, exponent=DEFAULT_EXPONENT, log_level="INFO"
+    nbits: int,
+    accurate: bool = True,
+    poolsize: int = 1,
+    exponent: int = DEFAULT_EXPONENT,
+    log_level: str = "INFO",
 ):
```
Comment (FoamyGuy) [inline adafruit_rsa/key.py]: I think we're missing the return type here right? 

Looks like maybe `Tuple["PublicKey", "PrivateKey"]`

File: adafruit_rsa/pkcs1.py [hunk 4]
```diff
@@ -229,15 +259,17 @@ def decrypt(crypto, priv_key):
     return cleartext[sep_idx + 1 :]
 
 
-def sign_hash(hash_value, priv_key, hash_method):
+def sign_hash(
+    hash_value: Optional[bytes], priv_key: PrivateKey, hash_method: str
```
Comment (FoamyGuy) [inline adafruit_rsa/pkcs1.py]: `hash_method` is perhaps another good candidate for Literal[]

`Literal['MD5', 'SHA-1', 'SHA-224', SHA-256', 'SHA-384', 'SHA-512']`

**Comment by FoamyGuy** [comment]
I found a few small changes to make. But vast majority of this is looking good to me. I tested all 3 example scripts successfully on a Feather TFT as well. Thanks @tekktrik !

File: adafruit_rsa/key.py (hunk 1 above)
Comment (tekktrik) [reply adafruit_rsa/key.py]: Yup!  Good catch!

File: adafruit_rsa/key.py (hunk 2 above)
Comment (tekktrik) [reply adafruit_rsa/key.py]: Yup!  Literal works for strings!

File: adafruit_rsa/key.py (hunk 3 above)
Comment (tekktrik) [reply adafruit_rsa/key.py]: Fixed!

File: adafruit_rsa/pkcs1.py (hunk 4 above)
Comment (tekktrik) [reply adafruit_rsa/pkcs1.py]: Ah yes, good catch!

**Comment by tekktrik** [comment]
Thanks for the review @FoamyGuy!  Changes made per review comments.

**Review by FoamyGuy — APPROVED** [review]
Looks good to me, thanks @tekktrik appreciate you knocking out this sizable one and all of the doc improvements!

**Comment by tekktrik** [comment]
No problem!  Thanks for the thorough read through!

**Pull request merged**